### PR TITLE
DOC-5285 - added topic on upgrading CorDapps

### PR DESCRIPTION
**Pull request by paulmoloneyr3**
Material from https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4459724853/Evolving+a+CorDapp - only material applicable to Corda 5.0 and not to any subsequent versions.

### CODE REVIEW

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    identifier: corda5-upgrading
+    parent: corda5-develop
+    weight: 9000
+section_menu: corda5
+
+---
+
+# Upgrading CorDapps
+
+This topic describes how to upgrade a Corda-deployed CorDapp to a newer version. This must be done in such a way that Corda will interpret the upgrade as a new version of the same application. Typically you might want to upgrade your CorDapp to fix bugs, create new functionality or change existing functionality.
```
Comment (colin-hillis) [inline content/en/platform/corda/5.0/developing-applications/upgrading/_index.md:17]:
```suggestion
This topic describes how to upgrade a Corda-deployed CorDapp to a newer version. This must be done in such a way that Corda will interpret the upgrade as a new version of the same application. Typically you might want to upgrade your CorDapp to fix bugs, create new functionality, or change existing functionality. 
```

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Corda must be able to interpret that the two versions are considered to be the same application. This is done in order to maintain continuity of any state it has preserved internally relating to that application's operation.
+
+Note that the scope of this topic does not include migrating application data persisted by any of the Corda persistence APIs. Nor does it include ensuring that different versions of flow implementations are compatible with one another.
+
+## Identifying CorDapps Across Versions
+
+Corda packages (CPKs) are identified across versions by the value of their `Corda CPK CorDapp Name` metadata field. Keeping the same value for `Corda CPK CorDapp Name` while making other changes to the CPK indicates to Corda that you consider the CPK to be a new version of the same CPK. As well as the value of  `Corda CPK CorDapp Name`, the set of signers of the CPK must also be kept the same across versions.
```
Comment (colin-hillis) [inline content/en/platform/corda/5.0/developing-applications/upgrading/_index.md:25]:
```suggestion
Corda packages (CPKs) are identified across versions by the value of their `Corda CPK CorDapp Name` metadata field. Keeping the same value for `Corda CPK CorDapp Name` while making other changes to the CPK indicates to Corda that you consider the CPK to be a new version of the same CPK. As well as the value of  `Corda CPK CorDapp Name`, the set of CPK signers must also be kept the same across versions.
```

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Any data internally serialized by Corda is tagged with both `Corda CPK CorDapp Name` and the set of signers that signed the CPK. On deserialization, the tag will be checked against the current version of the CPK’s `Corda CPK CorDapp Name` and set of signers. If both of these match, Corda will consider this data as deserializable for the CPK.
+
+How `Corda CPK CorDapp Name` is set differs between Corda 5.0 and subsequent versions.
+
+## Identifying CorDapps in Corda 5.0
+
+In Corda 5.0, `Corda CPK CorDapp Name` is not configurable, but is set using the concatenation of the following properties defined by the Corda build system:
```
Comment (colin-hillis) [inline content/en/platform/corda/5.0/developing-applications/upgrading/_index.md:33]:
```suggestion
In Corda 5.0, `Corda CPK CorDapp Name` is not configurable, but it is set using the concatenation of the following properties defined by the Corda build system:
```

**Review by colin-hillis — CHANGES_REQUESTED** [review]
Sorry just a few really minors things

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md (hunk 1 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/5.0/developing-applications/upgrading/_index.md]: Updated

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md (hunk 3 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/5.0/developing-applications/upgrading/_index.md]: Updated

File: content/en/platform/corda/5.0/developing-applications/upgrading/_index.md (hunk 2 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/5.0/developing-applications/upgrading/_index.md]: Updated

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by yarivus — COMMENTED** [review]
thx @paulmoloneyr3

**Pull request merged**